docs/guide/cli/migrate.md

```diff
@@ -0,0 +1,397 @@
+| `--manifest` | `string[]` | `[]` | Path(s) to custom migration manifest file(s) |
+
+## Examples
+
+### Basic Migration
+
+```sh
+# Migrate a single package
+e18e-cli migrate chalk
+
+# Migrate multiple packages
+e18e-cli migrate chalk lodash moment
+
+# Migrate all available packages
+e18e-cli migrate --all
+```
+
+### Advanced Options
+
+```sh
+# Preview changes without applying them
+e18e-cli migrate chalk --dry-run
+
+# Interactive mode to select packages
+e18e-cli migrate --interactive
+
+# Migrate specific file types
+e18e-cli migrate chalk --include "src/**/*.{ts,js,jsx,tsx}"
+
+# Combine options
+e18e-cli migrate --all --dry-run --include "src/**/*.ts"
+
+# Use custom migration manifest
+e18e-cli migrate --manifest ./custom-migrations.json
+```
+
+## Migration System
+
+### Opinionated Codemod
+
+The CLI uses e18e's **opinionated codemod** approach, which provides a curated list of well-tested, performance-focused package replacements. This ensures consistent migrations across the ecosystem and reduces the risk of breaking changes.
+
+> [!IMPORTANT]
+> The replacement list is still being developed and expanded. The current list represents the initial set of well-tested migrations. More replacements will be added as they are validated and tested across the ecosystem.
+
+### Default Migrations
+
+The CLI includes pre-configured migrations for common packages:
+
+### Performance Optimizations
+
+| From | To | Reason |
+|------|----|--------|
+| `chalk` | `picocolors` | Smaller, faster, no dependencies |
+| `lodash` | `lodash-es` | ESM version, better tree-shaking |
+| `moment` | `date-fns` | Tree-shakeable, smaller bundle |
+| `axios` | `undici` | Native fetch-like API, smaller |
+| `request` | `undici` | Modern HTTP client |
+
+### Modern Alternatives
+
+| From | To | Reason |
+|------|----|--------|
+| `uuid` | `crypto.randomUUID()` | Native Node.js API |
+| `left-pad` | `String.prototype.padStart()` | Native JavaScript |
+| `is-array` | `Array.isArray()` | Native JavaScript |
+
+### Custom Manifests
+
+You can extend or override the default migrations by providing custom manifest files. This allows you to:
+
+- Add your own package replacements
+- Override default migration rules
+- Create team-specific migration patterns
+- Test new migrations before contributing them back
+
+**Custom manifest format:**
+```json
+{
+  "replacements": [
+    {
+      "from": "old-package",
+      "to": "new-package",
+      "reason": "Better performance and smaller bundle",
+      "importMapping": {
+        "default": "named",
+        "named": ["function1", "function2"]
+      }
+    }
+  ]
+}
+```
+
+**Using custom manifests:**
+```sh
+# Single manifest file
+e18e-cli migrate --manifest ./my-migrations.json
+
+# Multiple manifest files
+e18e-cli migrate --manifest ./team-migrations.json --manifest ./project-migrations.json
+
+# Combine with other options
+e18e-cli migrate --manifest ./custom.json --dry-run --interactive
+```
+
+## How It Works
```

Comment:
probably don't need this section either (most of it is internals or common sense we don't need to document)